storage_service/src_08_01_2020/main/java/com/newgen/storage/service/impl/AzureStorageServiceImpl.java

```diff
@@ -0,0 +1,222 @@
+	private static final Logger logger = LoggerFactory.getLogger(AzureStorageServiceImpl.class);
+
+	static final String COMMAND_KEY_UPLOAD = "UploadCommandKey";
+
+	static final String COMMAND_KEY_DELETE = "DeleteCommandKey";
+
+	static final String DEFAULT_ENDPOINTS_PROTOCOL = "DefaultEndpointsProtocol=";
+
+	static final String ACCOUNT_NAME = ";AccountName=";
+
+	static final String ACCOUNT_KEY = ";AccountKey=";
+
+	@Value("${app.title}")
+	String appTitle;
+
+	@Value("${logging.service.url}")
+	private String loggingServiceUrl;
+
+	@Autowired
+	private RestTemplate restTemplate;
+
+	@HystrixCommand(fallbackMethod = "reliableUpload", commandKey = COMMAND_KEY_UPLOAD)
+	public String upload(String contentPath, StorageCredentials storageCredentials, String containerName)
+			throws Exception {
+		logger.debug("Uploading document to Azure Blob " + new Date());
+		CloudBlockBlob blob;
+		String storageConnectionString = DEFAULT_ENDPOINTS_PROTOCOL + storageCredentials.getStorageProtocol()
+				+ ACCOUNT_NAME + storageCredentials.getAccountName() + ACCOUNT_KEY + storageCredentials.getAccountKey();
+		logger.debug("Storage connection string: " + storageConnectionString);
+		// Retrieve storage account from connection-string.
+		CloudStorageAccount storageAccount = CloudStorageAccount.parse(storageConnectionString);
+		logger.debug("Retrieved storage account: " + storageAccount);
+		// Create the blob client.
+		CloudBlobClient blobClient = storageAccount.createCloudBlobClient();
+		blobClient.getDefaultRequestOptions().setTimeoutIntervalInMs(5000);
+		logger.debug("Blob client: " + blobClient);
+		// Retrieve reference to a previously created container.
+		CloudBlobContainer container;
+		container = blobClient.getContainerReference(containerName.toLowerCase());
+		logger.debug("Container: " + container.getName());
+		// Create the container if it does not exist.
+		// container.createIfNotExists();
+		boolean containerexist = container.createIfNotExists();
+		if (containerexist == true) {
+			callLoggingService(container.getName(), null, "Blob", null, null, null, null, "CREATE", "StorageService");
+		}
+		logger.debug("Container created");
+		logger.debug("contentPath: " + contentPath);
+		// Create filename to be uploaded on blob
+//		String ext = FilenameUtils.getExtension(contentPath);
+//		logger.debug("ext: " + ext);
+//		String fileName = FilenameUtils.getBaseName(contentPath);
+		String fileName = FilenameUtils.getName(contentPath);
+		logger.debug("fileName: " + fileName);
+//		String guid = UUID.randomUUID().toString();
+//		logger.debug("guid: " +guid);
+//		String uploadFileName = fileName + guid + "." + ext;
+//		logger.debug("Upload file name: " + uploadFileName);
+		blob = container.getBlockBlobReference(fileName);
+		logger.debug("step 1 " + blob);
+		File file = new File(contentPath);
+		logger.debug("step 2" + file);
+		FileInputStream fip = new FileInputStream(file);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *fip*. The resource is referenced in statements at the following line: 106. The resource closure statement is at line: 108. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by File.length. Either a) close *fip* in a try-finally block or b) close the resource by declaring *fip* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).